addons/calendar/models/calendar_event.py

```diff
@@ -397,6 +408,31 @@ def _compute_recurrence(self):
             else:
                 event.update(false_values)
 
+    @api.onchange('rrule_type_ui')
+    def _on_change_rrule_type_ui(self):
+        """ This field is only used in the UI to compute the recurrence settings
```

Comment:
We need to store the field so that when you open the form view again, it will show the correct configuration. Notice that, if I set a daily recurrence, the `rrule_type_ui` could be either 'daily' or 'custom'.

Reply:
This doesn't need to be stored. It can be determined from other recurrence fields.

Reply:
That won't happen. It will only show "custom" if it doesn't match a perfect daily/weekly/monthly recurrence. So you choose "daily", no reason that the auto-detection would then match "custom".